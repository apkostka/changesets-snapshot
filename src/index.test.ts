import * as core from '@actions/core';
import * as github from '@actions/github';
import { detect } from '@antfu/ni';

import { logger } from './logger';
import { run, runPublish } from './run';

import { publishSnapshot } from '.';

jest.mock('@actions/github');
jest.mock('@actions/core');
jest.mock('@antfu/ni', () => ({
  ...jest.requireActual('@antfu/ni'),
  detect: jest.fn().mockName('@antfu/ni.detect'),
}));
jest.mock('./npm-utils');
jest.mock('./run');
jest.mock('./logger');

const runMock = jest.mocked(run);
const runPublishMock = jest.mocked(runPublish);
const coreMock = jest.mocked(core, true);

const getScriptCalls = <T extends jest.MockableFunction>(
  mockedFn: jest.MockedFn<T>,
) => mockedFn.mock.calls.map((args) => args[0].script);

afterEach(() => {
  jest.clearAllMocks();
});

test.each(['yarn', 'npm', 'pnpm'])(
  'command output for %s',
  async (packageManager) => {
    process.env.GITHUB_TOKEN = '@github-token';
    process.env.NPM_TOKEN = '@npm-token';
    github.context.ref = 'feature/123-branch';
    runMock.mockResolvedValueOnce({
      code: 0,
      stdout: '',
      stderr: '',
    });
    runPublishMock.mockResolvedValueOnce({
      published: true,
      publishedPackages: [
        { name: '@multiple/package1', version: '1.2.3-SNAPSHOT' },
        { name: '@multiple/package-two', version: '1.2.3-SNAPSHOT' },
      ],
    });
    jest
      .mocked(detect)
      .mockResolvedValueOnce(
        packageManager as unknown as ReturnType<typeof detect>,
      );

    await publishSnapshot();

    expect(getScriptCalls(runMock)).toMatchSnapshot('run');
    expect(getScriptCalls(runPublishMock)).toMatchSnapshot('runPublish');
    expect(jest.mocked(logger).log.mock.calls[0]).toMatchSnapshot('logger.log');
    expect(coreMock.summary.addHeading.mock.calls[0][0]).toMatchSnapshot(
      'summary',
    );
    expect(coreMock.summary.addRaw.mock.calls[0][0]).toMatchSnapshot('summary');
    expect(coreMock.summary.addCodeBlock.mock.calls[0][0]).toMatchSnapshot(
      'summary',
    );
  },
);
