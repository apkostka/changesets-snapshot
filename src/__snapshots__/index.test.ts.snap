// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command output for npm: logger.log 1`] = `
Array [
  "2 snapshots published.
- @multiple/package1@1.2.3-SNAPSHOT
- @multiple/package-two@1.2.3-SNAPSHOT",
]
`;

exports[`command output for npm: run 1`] = `
Array [
  "node /__mocked_node_modules__/@changesets/cli/bin.js version --snapshot feature_123-branch",
]
`;

exports[`command output for npm: runPublish 1`] = `
Array [
  "node /__mocked_node_modules__/@changesets/cli/bin.js publish --tag feature_123-branch",
]
`;

exports[`command output for npm: summary 1`] = `"ðŸ¦‹ New snapshots published!"`;

exports[`command output for npm: summary 2`] = `"<p>Version: <code>@multiple/package1@1.2.3-SNAPSHOT</code></p>"`;

exports[`command output for npm: summary 3`] = `"npm i @multiple/package1@feature_123-branch"`;

exports[`command output for pnpm: logger.log 1`] = `
Array [
  "2 snapshots published.
- @multiple/package1@1.2.3-SNAPSHOT
- @multiple/package-two@1.2.3-SNAPSHOT",
]
`;

exports[`command output for pnpm: run 1`] = `
Array [
  "node /__mocked_node_modules__/@changesets/cli/bin.js version --snapshot feature_123-branch",
]
`;

exports[`command output for pnpm: runPublish 1`] = `
Array [
  "node /__mocked_node_modules__/@changesets/cli/bin.js publish --tag feature_123-branch",
]
`;

exports[`command output for pnpm: summary 1`] = `"ðŸ¦‹ New snapshots published!"`;

exports[`command output for pnpm: summary 2`] = `"<p>Version: <code>@multiple/package1@1.2.3-SNAPSHOT</code></p>"`;

exports[`command output for pnpm: summary 3`] = `"pnpm add @multiple/package1@feature_123-branch"`;

exports[`command output for yarn: logger.log 1`] = `
Array [
  "2 snapshots published.
- @multiple/package1@1.2.3-SNAPSHOT
- @multiple/package-two@1.2.3-SNAPSHOT",
]
`;

exports[`command output for yarn: run 1`] = `
Array [
  "node /__mocked_node_modules__/@changesets/cli/bin.js version --snapshot feature_123-branch",
]
`;

exports[`command output for yarn: runPublish 1`] = `
Array [
  "node /__mocked_node_modules__/@changesets/cli/bin.js publish --tag feature_123-branch",
]
`;

exports[`command output for yarn: summary 1`] = `"ðŸ¦‹ New snapshots published!"`;

exports[`command output for yarn: summary 2`] = `"<p>Version: <code>@multiple/package1@1.2.3-SNAPSHOT</code></p>"`;

exports[`command output for yarn: summary 3`] = `"yarn add @multiple/package1@feature_123-branch"`;
